src/Dscp/Educator/Web/Student/API.hs

```diff
@@ -0,0 +1,111 @@
+{-# LANGUAGE DataKinds     #-}
+{-# LANGUAGE TypeOperators #-}
+
+-- | Student HTTP API definition.
+
+module Dscp.Educator.Web.Student.API
+       ( API
+       ) where
+
+import Data.Time.Clock (UTCTime)
+import Servant.API
+import Servant.API.Verbs (StdMethod (..), Verb)
+
+import qualified Dscp.Core.Types as Core
+import Dscp.Crypto (Hash)
+import Dscp.Educator.Web.Student.Types (Assignment, BlkProof, Course, Submission)
+
+-- TODO: migrate to newer Servant which has `Summary` and `Description` combinators
+type API
+    =    GetCourses
+    :<|> GetCourse
+    :<|> GetAssignments
+    :<|> GetAssignment
+    :<|> GetSubmissions
+    :<|> GetSubmission
+    :<|> MakeSubmission
+    :<|> DeleteSubmission
+    :<|> GetProofs
+
+---------------------------------------------------------------------------
+-- Courses
+---------------------------------------------------------------------------
+
+type GetCourses
+    = "courses"
+    -- :> Summary "Get Educator's courses"
+    -- :> Description "Gets a list of Educator's courses, both enrolled and available."
+    :> QueryParam "enrolled" Bool
+    :> Verb 'GET 200 '[JSON] [Course]
+
+type GetCourse
+    = "courses" :> Capture "courseId" Int
+    -- :> Summary "Get info about the course"
+    -- :> Description "Gets all info about the given course."
+    :> Verb 'GET 200 '[JSON] Course
+
+---------------------------------------------------------------------------
+-- Assignments
+---------------------------------------------------------------------------
+
+type GetAssignments
+    = "assignments"
+    -- :> Summary "Get student's assignments"
+    -- :> Description "Gets a list of student's assignments. Filter parameters are \
+                   -- \used to specify specific course, type, etc."
+    :> QueryParam "course" Core.Course
+    :> QueryParam "type" Core.DocumentType
+    :> QueryParam "final" Bool
```

Comment:
I'm not sure whether my comment to swagger PR was published
It seems to me that having 'Bool' here is not good because you can not add new assignment types in a backward-compatible way.